### ISSUE TITLE
Clean up config parameter names

### ISSUE THREAD

**Issue by Janekdererste** [matsim-org/matsim-libs#1276]
> More difficult: How to deal with the controler module in the xml config. Suggestions? Maybe we could no longer accept the old spelling and write a meaningful error message.

the config could use some name-clean-up anyway (e.g. `planscalcroute`, `planCalcScore`). I already thought about adding an "alias-list" to the config-reader: We could rename the modules internally, so they will be written out in the new names, but when loading, we could do a simple lookup. It would be something like a map:

```java
Map<String, String> aliases = new HashMap<>();
aliases.put("controler", "controller");
aliases.put("planCalcScore", "scoring");
aliases.put("plancalcroute", "routing");

...
moduleName = aliases.getOrDefault(moduleName, moduleName);
...
```
One could even check if the name has changed and issue a warning in this case.

If this is of interest, we probably should discuss this in a different issue, including how the names are going to be cleaned up.

_Originally posted by @mrieser in https://github.com/matsim-org/matsim-libs/issues/1273#issuecomment-737885391_

**Comment by mrieser** [comment]
#2473 provides the infrastructure for renaming config modules and parameters.

I have not yet done any renaming in the config groups, I would wait until the 15.0 release was created and only start renaming things afterwards for 16.0-SNAPSHOT, so we have enough time to update all the example files and docs.

**Comment by mrieser** [comment]
## Status Quo

The current config module names and parameter names in the MATSim config
are a wild collection without recognizable scheme:

Module names:

- ReplanningAnnealer
- TimeAllocationMutator
- changeMode
- parallelEventHandling
- planCalcScore
- planscalcroute

Parameter names:
- mutationRange
- dumpDataAtEnd
- BrainExpBeta
- PathSizeLogitBeta
- marginalUtilityOfDistance_util_m

Parameterset types:
- scoringParameters
- activityParams
- modeParams
- teleportedModeParameters


## Proposal: Naming Convention

All module names, parameter names and parameterset types follow the naming convention for Java class members or variables:

- start with a small letter
- use camelCase (capital letters do denote new words, no hyphens - or underscores _)

One could argue that module names and parameterset types are more like classes and should start with a capital letter. But besides a few exceptions, most module names currently start with small letters, so I would keep them (think of `global`, `network`, etc, which all would have to be changed if capital letters should be used).

## Other proposed renamings

- controler &rarr; controller
- planCalcScore &rarr; scoring
- planscalcroute &rarr; routing
- strategy &rarr; replanning
- parallelEventHandling &rarr; eventsManager

When renaming module names, the corresponding Java classes should be renamed as well (e.g. `ReplanningConfigGroup`, `EventsManagerConfigGroup` etc)

**Comment by kainagel** [comment]
What about units?  There is one or two places where units are added: "_utils_m".  Thus is ugly, but we know from teaching and support how often this goes wrong.  This implies to accept the ugliness and rather add units here and there than remove them.  --  I do not mind if these become underscored or camel case, but I think that underscores are easier to read: marginalUtilityOfDistanceUtilM vs marginalUtilityOfDistance_util_m .  So I would vote for underscores for units, despite the fact that this is even more ugly.  Form follows function. :-|

I explicitly support making names in code and names in xml config the same.

**Comment by mrieser** [comment]
sorry, yes, units should be included using `_`. I had this written down in another place but forgot to move it over here. I will edit the description above.

**Comment by mrieser** [comment]
Closed by #2852
